### PR TITLE
Streamline build process to use a single make and config file

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by isbadawi** [comment]
LGTM. You should be able to delete `get_messenger_dir.py`, no?

File: messenger/make.py [hunk 1]
```diff
@@ -1,71 +1,99 @@
...
-os.system(make_cmd)
-
-messenger_exe = 'messenger.%s'%extension
-messenger_loc = os.path.join(messenger_dir, messenger_exe)
-
-shutil.move(messenger_exe, messenger_loc)
 
+if __name__ == '__main__':
+    usage = 'Please specify a valid make target (Matlab or Octave)'
+    if len(sys.argv) < 1:
```
Comment (isbadawi) [inline messenger/make.py]: Should be 2, not 1

File: messenger/make.py [hunk 2]
```diff
@@ -1,71 +1,99 @@
...
+    extension = check_extension.stdout.read()
+    extension = extension.decode('utf-8').rstrip('\r\n')
+
+    # Build the mex file
+    if sys.platform == 'win32':
+        mex = matlab_bin + "\\mex.bat"
+    else:
+        mex = matlab_bin + "/mex"
+    paths = "-L%(zmq_lib)s -I%(zmq_inc)s" % cfg
+    make_cmd = '"%s" -O %s -lzmq ./src/messenger.c' % (mex, paths)
```
Comment (isbadawi) [inline messenger/make.py]: With the quotes around the executable I get an `OSError: [Errno 2] No such file or directory` on my mac.

Comment (blink1073) [inline messenger/make.py]: Hmm, looking at the original version it looks like those quotes were there.  But yeah I got the same error on Linux with Matlab (must be a windows-only thing).

Comment (isbadawi) [inline messenger/make.py]: Probably just `os.system` has different semantics than `check_output`.

Comment (blink1073) [inline messenger/make.py]: Ah, you reminded me to use `shlex.split`, which will handle spaces in paths.

Comment (blink1073) [inline messenger/make.py]: And yes, that does make sense, since we are passing a list of strings.

File: messenger/make.py [hunk 3]
```diff
@@ -1,71 +1,104 @@
...
+    subprocess.check_output(shlex.split(make_cmd))
+
+    messenger_loc = os.path.join(messenger_dir, messenger_exe)
+
+    shutil.move(messenger_exe, messenger_loc)
+
+    try:
+        os.remove('messenger.o')
+    except OSError:
+        os.remove('src/messenger.o')
```
Comment (isbadawi) [inline messenger/make.py]: I still get an error here when building for matlab. I ran the mex command manually and it looks like I don't get a `.o` file anywhere, actually.

**Pull request merged**